decidim-core/app/packs/src/decidim/editor/common/input_dialog.js

```diff
@@ -0,0 +1,214 @@
+import Dialog from "a11y-dialog-component";
+
+import { getDictionary } from "src/decidim/i18n";
+import { uniqueId } from "src/decidim/editor/common/helpers";
+
+export default class InputDialog {
+  constructor(editor, { inputs }) {
+    this.editor = editor;
+    // The legacy design should not have any elements on the page with the
+    // `data-dialog` attribute.
+    this.legacyDesign = !document.querySelector("[data-dialog]");
+    const inputId = uniqueId("inputdialog");
+    this.element = document.createElement("div");
+    if (this.legacyDesign) {
+      this.element.classList.add("reveal");
+      this.element.setAttribute("data-reveal", "");
+    } else {
+      this.element.dataset.dialog = `${Math.random().toString(36).slice(2)}`;
+    }
+
+    let inputsHTML = "";
+    Object.keys(inputs).forEach((name) => {
+      const input = inputs[name];
+      let inputHTML = "";
+      if (input.type === "select") {
+        const optionsHTML = input.options.map((opt) => `<option value="${opt.value}">${opt.label}</option>`)
+        inputHTML = `<select id="${inputId}-${name}">${optionsHTML}</select>`;
+      } else {
+        inputHTML = `<input id="${inputId}-${name}" type="${input.type || "text"}">`;
+      }
+
+      inputsHTML += `
+        <div data-input="${name}">
+          <label>
+            ${input.label}
+            ${inputHTML}
+          </label>
+        </div>
+      `;
+    });
+
+    const i18n = getDictionary("editor.inputDialog");
+
+    let buttonsHTML = "";
+    if (this.legacyDesign) {
```

Comment:
Why do we have a check here, to treat the legacy Design, yet  exactly 8 lines lower you do the same thing ? Could we have one single check combining the 2 ?

Reply:
Good point, there was something else here before which is why it was like this.

Fixed at 9187ee5c3005d7d36574878f11642d73e7b1af91.